<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from raven_2arm.xacro               | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<!--RavenII Surgical Robot 2-Arm urdf File -->
<!--Author: Sina Nia Kosari & Lee White-->
<!--kosari@uw.edu & leewhite@uw.edu-->
<!--Date: 01-Dec-2011 -->
<!--Biorobotics Laboratory -->
<!--University of Washington -->
<!--Changed by Yongming-->
<!--03-2018-->
<!-- For gazebo. Yongming. 2018/11/15 -->
<robot name="raven_2arm" xmlns:xacro="http://www.ros.org/wiki/xacro">
  
  
  <!--This is a virtual link, both robot bases are attached to this link -->
  <link name="world"/>
  <!-- This is the steel frame. There is a joint setting in joint part -->
  <link name="frame">
    <visual>
      <geometry>
        <mesh filename="package://my_urdf/Raven_files/steel_frame.STL" scale="0.01 0.01 0.01"/>
      </geometry>
      <origin rpy="1.57075 0 3.1415" xyz="0 0 0"/>
      <material name="white">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://my_urdf/Raven_files/steel_frame.STL" scale="0.01 0.01 0.01"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.25"/>
      <inertia ixx="0.0416666666667" ixy="0.0" ixz="0.0" iyy="0.0416666666667" iyz="0.0" izz="0.0416666666667"/>
    </inertial>
  </link>
  <!-- joint for frame. the values of z are all reassigned to avoid collision after small movement -->
  <joint name="frame_joint" type="fixed">
    <!-- <origin rpy="0 0 0" xyz="0.245 -0.265 -0.235"/> contain frame's dimentsion -->
    <origin rpy="0 0 0" xyz="0.245 -0.265 0"/>
    <parent link="world"/>
    <child link="frame"/>
  </joint>
  <!-- LEFT ARM LINKS -->
  <link name="base_link_L">
    <visual>
      <geometry>
        <mesh filename="package://my_urdf/Raven_files/left_arm/base_w_motors.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://my_urdf/Raven_files/left_arm/base_w_motors.dae"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.25"/>
      <inertia ixx="0.0416666666667" ixy="0.0" ixz="0.0" iyy="0.0416666666667" iyz="0.0" izz="0.0416666666667"/>
    </inertial>
  </link>
  <!-- LEFT ARM JOINTS -->
  <joint name="camera_L" type="fixed">
    <limit effort="1000.0" lower="0.0" upper="3.1415" velocity="0.5"/>
    <!-- original file has this two option and chooses the second one. ming-->
    <origin rpy="0 0 0" xyz="-0.2 0 0.235"/>
    <parent link="world"/>
    <child link="base_link_L"/>
  </joint>
  <link name="link1_L">
    <visual>
      <geometry>
        <mesh filename="package://my_urdf/Raven_files/left_arm/link1_w_cover.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://my_urdf/Raven_files/left_arm/link1_w_cover.dae"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.503"/>
      <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.000001" iyz="0.0" izz="0"/>
    </inertial>
  </link>
  <joint name="shoulder_L" type="revolute">
    <axis xyz="-1 0 0"/>
    <limit effort="1000.0" lower="0.1" upper="1.396" velocity="0.1"/>
    <!-- 15tb 80 degree -->
    <!-- <origin rpy="0 1.5705 0" xyz="0.0 0.0 0"/> -->
    <parent link="base_link_L"/>
    <child link="link1_L"/>
    <dynamics damping="0.7"/>
  </joint>
  <link name="link2_L">
    <visual>
      <geometry>
        <mesh filename="package://my_urdf/Raven_files/left_arm/link2.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://my_urdf/Raven_files/left_arm/link2.dae"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.25"/>
      <inertia ixx="0.0416666666667" ixy="0.0" ixz="0.0" iyy="0.0416666666667" iyz="0.0" izz="0.0416666666667"/>
    </inertial>
  </link>
  <joint name="elbow_L" type="revolute">
    <axis xyz="-0.2590 0.9014 -0.3471"/>
    <limit effort="1000.0" lower="0.1745" upper="2.0944" velocity="0.1"/>
    <!-- 10 120 degree -->
    <!-- <origin rpy="0 0 0" xyz="0.0 0.0 0"/> -->
    <parent link="link1_L"/>
    <child link="link2_L"/>
    <dynamics damping="0.7"/>
  </joint>
  <link name="link3_L">
    <visual>
      <geometry>
        <mesh filename="package://my_urdf/Raven_files/left_arm/link3.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://my_urdf/Raven_files/left_arm/link3.dae"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.025"/>
      <inertia ixx="0.0208333333333" ixy="0.0" ixz="0.0" iyy="0.0208333333333" iyz="0.0" izz="0.0208333333333"/>
    </inertial>
  </link>
  <joint name="insertion_L" type="prismatic">
    <axis xyz="-0.6012 -0.7460 0.2864"/>
    <limit effort="1000.0" lower="-0.2" upper="0.05" velocity="0.01"/>
    <!-- -.17 .1 original -->
    <!-- <origin rpy="0 0 0" xyz="0 0 0"/> -->
    <parent link="link2_L"/>
    <child link="link3_L"/>
    <dynamics damping="0.7"/>
  </joint>
  <link name="instrument_shaft_L">
    <visual>
      <geometry>
        <mesh filename="package://my_urdf/Raven_files/left_arm/instrument_shaft.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://my_urdf/Raven_files/left_arm/instrument_shaft.dae"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.0125"/>
      <inertia ixx="0.0104166666667" ixy="0.0" ixz="0.0" iyy="0.0104166666667" iyz="0.0" izz="0.0104166666667"/>
    </inertial>
  </link>
  <joint name="tool_roll_L" type="revolute">
    <axis xyz="-0.6012 -0.7460 0.2864"/>
    <limit effort="1000.0" lower="-3.1415" upper="3.1415" velocity="0.5"/>
    <!-- -180 180 degree -->
    <!-- <origin rpy="0 0 0" xyz="0 0 0"/> -->
    <parent link="link3_L"/>
    <child link="instrument_shaft_L"/>
    <dynamics damping="0.7"/>
  </joint>
  <link name="wrist_L">
    <visual>
      <geometry>
        <mesh filename="package://my_urdf/Raven_files/left_arm/wrist.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://my_urdf/Raven_files/left_arm/wrist.dae"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.0025"/>
      <inertia ixx="0.00520833333333" ixy="0.0" ixz="0.0" iyy="0.00520833333333" iyz="0.0" izz="0.00520833333333"/>
    </inertial>
  </link>
  <joint name="wrist_joint_L" type="revolute">
    <axis xyz="-0.5637 0.6523 0.5067"/>
    <limit effort="1000.0" lower="-1.39626" upper="1.39626" velocity="0.5"/>
    <!-- -80 80 degree -->
    <!-- <origin rpy="0 0 0" xyz="0 0 0"/> -->
    <parent link="instrument_shaft_L"/>
    <child link="wrist_L"/>
    <dynamics damping="0.7"/>
  </joint>
  <link name="grasper1_L">
    <visual>
      <geometry>
        <mesh filename="package://my_urdf/Raven_files/left_arm/grasper_1.dae"/>
      </geometry>
      <origin rpy="0 0 0" xyz="-0.00680 -0.00904 0.00452"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="-0.00680 -0.00904 0.00452"/>
      <geometry>
        <mesh filename="package://my_urdf/Raven_files/left_arm/simple_ravenII_dae/grasper_1.dae"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="-0.00680 -0.00904 0.00452"/>
      <mass value="0.0025"/>
      <inertia ixx="0.00260416666667" ixy="0.0" ixz="0.0" iyy="0.00260416666667" iyz="0.0" izz="0.00260416666667"/>
    </inertial>
  </link>
  <joint name="grasper_joint_1_L" type="revolute">
    <axis xyz="-0.5648 0.1427 -0.8128"/>
    <limit effort="1000.0" lower="-1.39626" upper="1.39626" velocity=".5"/>
    <!-- -80 80 degree -->
    <origin rpy="0 0 0" xyz="0.00680 0.00904 -0.00452"/>
    <parent link="wrist_L"/>
    <child link="grasper1_L"/>
    <dynamics damping="0.7"/>
  </joint>
  <!-- comment for inverse kinematics, otherwise the end-effector is not easy to be decided 
  <link name="grasper2_L">
    <visual>
      <geometry>
        <mesh filename="package://my_urdf/Raven_files/left_arm/grasper_2.dae"/>
      </geometry>
      <origin rpy="0 0 0" xyz="-0.00680 -0.00904 0.00452"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="-0.00680 -0.00904 0.00452"/>
      <geometry>
        <mesh filename="package://my_urdf/Raven_files/left_arm/simple_ravenII_dae/grasper_2.dae"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="-0.00680 -0.00904 0.00452"/>
      <mass value="0.0025"/>
      <inertia ixx="0.00260416666667" ixy="0.0" ixz="0.0" iyy="0.00260416666667" iyz="0.0" izz="0.00260416666667"/>
    </inertial>
  </link>
  <joint name="grasper_joint_2_L" type="revolute">
    <axis xyz="-0.5648 0.1427 -0.8128"/>
    <limit effort="1000.0" lower="-1.39626" upper="1.39626" velocity=".5"/>
    --> <!-- -80 80 degree --> <!--
    <origin rpy="0 0 0" xyz="0.00680 0.00904 -0.00452"/>
    <parent link="wrist_L"/>
    <child link="grasper2_L"/>
    <dynamics damping="0.7"/>
  </joint> -->

  <!-- RIGHT ARM LINKS -->
  <link name="base_link_R">
    <visual>
      <origin rpy="3.1415 -1.57075 0" xyz="0 0 1"/>
      <geometry>
        <mesh filename="package://my_urdf/Raven_files/right_arm/base_w_motors.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="3.1415 -1.57075 0" xyz="0 0 1"/>
      <geometry>
        <mesh filename="package://my_urdf/Raven_files/right_arm/base_w_motors.dae"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="3.1415 -1.57075 0" xyz="0 0 1"/>
      <mass value="0.25"/>
      <inertia ixx="0.0416666666667" ixy="0.0" ixz="0.0" iyy="0.0416666666667" iyz="0.0" izz="0.0416666666667"/>
    </inertial>
  </link>
  <joint name="camera_R" type="fixed">
    <limit effort="1000.0" lower="0.0" upper="3.1415" velocity="0.5"/>
    <origin rpy="3.1415 -1.57075 0" xyz="-0.33 0.00 0.235"/>
    <parent link="world"/>
    <child link="base_link_R"/>
    <dynamics damping="0.7"/>
  </joint>
  <link name="link1_R">
    <visual>
      <origin rpy="0 0 0" xyz="1 0 0"/>
      <geometry>
        <mesh filename="package://my_urdf/Raven_files/right_arm/link1_w_cover.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="1 0 0"/>
      <geometry>
        <mesh filename="package://my_urdf/Raven_files/right_arm/link1_w_cover.dae"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="1 0 0"/>
      <mass value="0.503"/>
      <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.000001" iyz="0.0" izz="0"/>
    </inertial>
  </link>
  <joint name="shoulder_R" type="revolute">
    <axis xyz="-1 0 0"/>
    <limit effort="1000.0" lower="0.1" upper="1.5708" velocity="0.5"/>
    <!-- 25tb 90 degree -->
    <!--origin rpy="0 1.5705 0" xyz="0 0.0 0"/-->
    <origin rpy="3.1415 -1.57075 0" xyz="0 0.0 0"/>
    <parent link="base_link_R"/>
    <child link="link1_R"/>
    <dynamics damping="0.7"/>
  </joint>
  <link name="link2_R">
    <visual>
      <origin rpy="0 0 0" xyz="1 0 0"/>
      <geometry>
        <mesh filename="package://my_urdf/Raven_files/right_arm/link2.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="1 0 0"/>
      <geometry>
        <mesh filename="package://my_urdf/Raven_files/right_arm/link2.dae"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="1 0 0"/>
      <mass value="0.25"/>
      <inertia ixx="0.0416666666667" ixy="0.0" ixz="0.0" iyy="0.0416666666667" iyz="0.0" izz="0.0416666666667"/>
    </inertial>
  </link>
  <joint name="elbow_R" type="revolute">
    <axis xyz="-0.2590 -0.9014 0.3471"/>
    <limit effort="1000.0" lower="0.1745" upper="2.0944" velocity="0.5"/>
    <!-- 10 120 degree -->
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="link1_R"/>
    <child link="link2_R"/>
    <dynamics damping="0.7"/>
  </joint>
  <link name="link3_R">
    <visual>
      <origin rpy="0 0 0" xyz="1 0 0"/>
      <geometry>
        <mesh filename="package://my_urdf/Raven_files/right_arm/link3.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="1 0 0"/>
      <geometry>
        <mesh filename="package://my_urdf/Raven_files/right_arm/link3.dae"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="1 0 0"/>
      <mass value="0.025"/>
      <inertia ixx="0.0208333333333" ixy="0.0" ixz="0.0" iyy="0.0208333333333" iyz="0.0" izz="0.0208333333333"/>
    </inertial>
  </link>
  <joint name="insertion_R" type="prismatic">
    <axis xyz="0.6012   -0.7460    0.2864"/>
    <limit effort="1000.0" lower="-.15" upper=".08" velocity="0.5"/>
    <!-- -.17 .1 original -->
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="link2_R"/>
    <child link="link3_R"/>
    <dynamics damping="0.7"/>
  </joint>
  <link name="instrument_shaft_R">
    <visual>
      <origin rpy="0 0 0" xyz="1 0 0"/>
      <geometry>
        <mesh filename="package://my_urdf/Raven_files/right_arm/instrument_shaft.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="1 0 0"/>
      <geometry>
        <mesh filename="package://my_urdf/Raven_files/right_arm/instrument_shaft.dae"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="1 0 0"/>
      <mass value="0.0125"/>
      <inertia ixx="0.0104166666667" ixy="0.0" ixz="0.0" iyy="0.0104166666667" iyz="0.0" izz="0.0104166666667"/>
    </inertial>
  </link>
  <joint name="tool_roll_R" type="revolute">
    <axis xyz="-0.6012 0.7460 -0.2864"/>
    <limit effort="1000.0" lower="-3.1415" upper="3.1415" velocity="0.5"/>
    <!-- -180 180 degree -->
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="link3_R"/>
    <child link="instrument_shaft_R"/>
    <dynamics damping="0.7"/>
  </joint>
  <link name="wrist_R">
    <visual>
      <origin rpy="0 0 0" xyz="1 0 0"/>
      <geometry>
        <mesh filename="package://my_urdf/Raven_files/right_arm/wrist.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="1 0 0"/>
      <geometry>
        <mesh filename="package://my_urdf/Raven_files/right_arm/wrist.dae"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="1 0 0"/>
      <mass value="0.0025"/>
      <inertia ixx="0.00520833333333" ixy="0.0" ixz="0.0" iyy="0.00520833333333" iyz="0.0" izz="0.00520833333333"/>
    </inertial>
  </link>
  <joint name="wrist_joint_R" type="revolute">
    <axis xyz=" 0.5637    0.6523    0.5067"/>
    <limit effort="1000.0" lower="-1.39626" upper="1.39626" velocity="0.5"/>
    <!-- -80 80 degree -->
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="instrument_shaft_R"/>
    <child link="wrist_R"/>
    <dynamics damping="0.7"/>
  </joint>
  <link name="grasper1_R">
    <visual>
      <origin rpy="0 0 0" xyz="1.00680 -0.00904 0.00452"/>
      <geometry>
        <mesh filename="package://my_urdf/Raven_files/right_arm/grasper_1.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="1.00680 -0.00904 0.00452"/>
      <geometry>
        <mesh filename="package://my_urdf/Raven_files/right_arm/grasper_1.dae"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="1.00680 -0.00904 0.00452"/>
      <mass value="0.0025"/>
      <inertia ixx="0.00260416666667" ixy="0.0" ixz="0.0" iyy="0.00260416666667" iyz="0.0" izz="0.00260416666667"/>
    </inertial>
  </link>
  <joint name="grasper_joint_1_R" type="revolute">
    <axis xyz="0.5648 -0.1427 -0.8128"/>
    <limit effort="1000.0" lower="-1.39626" upper="1.39626" velocity=".5"/>
    <!-- -80 80 degree -->
    <origin rpy="0 0 0" xyz="-0.00680 0.00904 -0.00452"/>
    <parent link="wrist_R"/>
    <child link="grasper1_R"/>
    <dynamics damping="0.7"/>
  </joint>
  <!-- same as left arm 
  <link name="grasper2_R">
    <visual>
      <origin rpy="0 0 0" xyz="1.00680 -0.00904 0.00452"/>
      <geometry>
        <mesh filename="package://my_urdf/Raven_files/right_arm/grasper_2.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="1.00680 -0.00904 0.00452"/>
      <geometry>
        <mesh filename="package://my_urdf/Raven_files/right_arm/grasper_2.dae"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="1.00680 -0.00904 0.00452"/>
      <mass value="0.0025"/>
      <inertia ixx="0.00260416666667" ixy="0.0" ixz="0.0" iyy="0.00260416666667" iyz="0.0" izz="0.00260416666667"/>
    </inertial>
  </link>
  <joint name="grasper_joint_2_R" type="revolute">
    <axis xyz="0.5648    0.1427   -0.8128"/>
    <limit effort="1000.0" lower="-1.39626" upper="1.39626" velocity=".5"/>
    --> <!-- -80 80 degree --> <!-- 
    <origin rpy="0 0 0" xyz="-0.00680 0.00904 -0.00452"/>
    <parent link="wrist_R"/>
    <child link="grasper2_R"/>
    <dynamics damping="0.7"/>
  </joint> -->

  <transmission name="tran1">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="shoulder_L">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor1">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="tran2">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="elbow_L">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor2">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="tran3">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="insertion_L">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor3">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="tran4">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="tool_roll_L">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor4">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="tran5">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wrist_joint_L">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor5">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="tran6">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="grasper_joint_1_L">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor6">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="tran7">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="grasper_joint_2_L">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor7">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="tran8">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="shoulder_R">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor8">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="tran9">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="elbow_R">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor9">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="tran10">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="insertion_R">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor10">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="tran11">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="tool_roll_R">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor11">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="tran12">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wrist_joint_R">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor12">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="tran13">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="grasper_joint_1_R">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor13">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="tran14">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="grasper_joint_2_R">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor14">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
</robot>

